src/main/java/football/pages/SantanderResultsPage.java

```diff
@@ -0,0 +1,79 @@
+    private List<WebElement> matchesListElement;
+    private By matchDiv = By.cssSelector("div[class='styled__GameweekTitle-qe32sx-2 jFqvgQ']");
+    private By resultsContainerLocator = By.cssSelector("div[class='show styled__ShowLoadedContainer-sc-191gwr0-0 khRXzW']");
+
+    public SantanderResultsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public String[] getTheCurrentMatchdayMatch(String teamName){
+        WebDriverWait ewait1 = new WebDriverWait(driver,10);
+        ewait1.until(ExpectedConditions.visibilityOfElementLocated(resultsContainerLocator));
+
+        matchesListElement = findElements(By.cssSelector("div[class='styled__ResultContainer-qe32sx-6 iYHKYB']"));
+        String [] match = new String[3];
+
+        for (int i = 0; i < matchesListElement.size(); i++) {
+            List<WebElement> eachMatch = matchesListElement.get(i).findElements(By.tagName("p"));
+            if(getText(eachMatch.get(0)).contains(teamName.toUpperCase()) && !getText(eachMatch.get(1)).equals("VS")){
+                match[0]  = getText(eachMatch.get(0));
+                match[1] = getText(eachMatch.get(1));
+                match[2] = getText(eachMatch.get(2));
+
+                break;
+            }
+            if (getText(eachMatch.get(2)).contains(teamName.toUpperCase()) && !getText(eachMatch.get(1)).equals("VS")) {
+                match[0] = getText(eachMatch.get(0));
+                match[1] = getText(eachMatch.get(1));
+                match[2] = getText(eachMatch.get(2));
+
+                break;
+            }
+            if(getText(eachMatch.get(1)).equals("VS")){
+                break;
+            }
+        }
+        return match;
+    }
+
+    public void moveToPreviousMatchday(){
+        matchdaysDropdownElement = findElements(By.cssSelector("div[class='styled__DropDownContainer-sc-1yaayks-1 gECFwF']"));
+        click(matchdaysDropdownElement.get(1));
+        String matchday = getText(matchdaysDropdownElement.get(1).findElement(By.tagName("p")));
+        String[] data = matchday.split(" ", 2);
+        int matchdayNumber = Integer.parseInt(data[1]);
+        List<WebElement> matchdaysList = matchdaysDropdownElement.get(1).findElement(By.tagName("ul")).findElements(By.tagName("li"));
+        String lastMatchday = "Jornada 1";
+        if (!matchday.equals(lastMatchday)){
+            click(matchdaysList.get(matchdayNumber-2));
+
+            WebDriverWait ewait1 = new WebDriverWait(driver,10);
+            ewait1.until(ExpectedConditions.visibilityOfElementLocated(resultsContainerLocator));
+        }
+    }
+
+    public int getMatchdaysAmount(){
+        WebDriverWait ewait1 = new WebDriverWait(driver,10);
+        ewait1.until(ExpectedConditions.visibilityOfElementLocated(matchDiv));
+
+        matchdaysDropdownElement = findElements(By.cssSelector("div[class='styled__DropDownContainer-sc-1yaayks-1 gECFwF']"));
+
+        String matchday = getText(matchdaysDropdownElement.get(1));
```

Comment:
Not how this part works but probably you can get the number MatchDay with the size of some list instead of parsing the data

Reply:
This method will return the matchdays amount counting from the default one. For example, once you move to Santander league in this week the default match day is 5, even when there are 6 matchdays total. The ones that actually care are the one by default and the previous ones right?
Maybe I could give the matchday amount by the size of this list: matchdaysDropdownElement = findElements(By.cssSelector("div[class='styled__DropDownContainer-sc-1yaayks-1 gECFwF']"));
But I'm not sure if the amount of matchdays is always the one by default + 1